frontend/src/components/shared/userCard/UserCard.test.tsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import UserCard from './UserCard';
+import { render } from '@testing-library/react';
+import { IUser } from '../../interfaces/Interface';
+import { shallow, configure, mount } from 'enzyme';
+import Adapter from 'enzyme-adapter-react-16';
+import { getUserFromStorage } from '../../../services/sessionStorageService';
+import { MemoryRouter } from 'react-router-dom';
+import { addFriendToDB } from '../../../services/apiUserService';
+import { createDialog } from '../../../services/apiChatService';
+
+configure({ adapter: new Adapter() });
+
+const mockHistoryPush = jest.fn();
+
+jest.mock('../../../services/apiUserService', () => ({
+  addFriendToDB: jest.fn().mockResolvedValue('SUCCESS')
+}));
+
+jest.mock('../../../services/apiChatService', () => ({
+  createDialog: jest.fn().mockResolvedValue(null)
+}));
+
+jest.mock('../../../services/sessionStorageService', () => ({
+  getUserFromStorage: jest.fn().mockReturnValue({ _id: 'abc123' }).mockReturnValueOnce(null)
+}));
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useHistory: () => ({
+    push: mockHistoryPush
+  })
+}));
+
+const userProps: IUser = {
+  _id: '5fae8f5bac677440d079a289',
+  about: 'hello',
+  avatar: 'abc',
+  birthday: new Date('1997-01-31T00:00:00.000Z'),
+  country: 'Ukraine',
+  countryCode: 'UK',
+  firstName: 'Serhii',
+  lastName: 'Kaliuzhnyi',
+  learn: [
+    {
+      language: 'Ukrainian',
+      level: 'Native'
+    }
+  ],
+  speak: [
+    {
+      language: 'English',
+      level: 'Intermediate'
+    }
+  ]
+};
+
+describe('UserCard test', () => {
+  beforeEach(() => {
+    render(
+      <MemoryRouter>
+        <UserCard mainInfo={userProps} />
+      </MemoryRouter>
+    );
+  });
+
+  it('Should redirect to / if user doesn`t exists in sessionStorage', () => {
+    expect(getUserFromStorage).toReturnWith(null);
+    //TODO check redirect to login if null
+  });
+
+  it('Should check links to profilePage are available', () => {
+    const wrapper = mount(
+      <MemoryRouter>
+        <UserCard mainInfo={userProps} />
+      </MemoryRouter>
+    );
+    const links = wrapper.find('.leftbar[href="/user/5fae8f5bac677440d079a289"]').length;
+    expect(links).toBeGreaterThan(0);
```

Comment:
maybe .toBeEqual(1)?

Reply:
also it is better to search for links in wrapper and then check that href prop is equal to required